kore/src/Kore/Internal/Pattern.hs

```diff
@@ -207,3 +209,21 @@ toPredicate = Conditional.toPredicate
 
 splitTerm :: Pattern variable -> (TermLike variable, Predicate variable)
 splitTerm = Conditional.splitTerm
+
+unparsePattern
```

Comment:
You can use this as the definition of `instance Unparse (Pattern variable)`, and avoid most of the other code changes. It would also be good to have it use `Kore.Internal.TermLike.externalizeFreshVariables`.